### DISCUSSION TITLE
Add scalable icons to categories in Adwaita.

### DISCUSSION THREAD

**Discussion by Initu-Castilhos** [linuxmint/discussions#121] [Ideas]
Other distributions like EndevouOS, Manjaro and Arch Adwaita lack many icons. I know it's the responsibility of the GNOME team and Linux Mint maintains icon version 41, but I reported this issue and they told me that the DE developers should address this issue. The solution would be to include scalable category icons and generic icons in apps with missing icons in the hicolor icon folder. The example of the problem shows the image below:
![227735290-fd85f5db-fedb-45a7-a488-0093885fb56d](https://github.com/linuxmint/discussions/assets/59381215/e5df12e1-2962-41b5-a038-05857626fc37)

Put a different way, newer versions of adwaita-icon-theme have removed a number of what they consider "legacy" icons, starting around version 41. Since then, Debian had been maintaining a 'patched' version of their package to restore these removed icons, and other distros had followed suit. However it doesn't look like they've published updated packages for version 44.

Gentoo just stabilized and switched to version 44 without these "legacy" icons present.
![236732192-a0340cee-fabb-46c6-bada-475cab7b0727](https://github.com/linuxmint/discussions/assets/59381215/cc154560-352d-493a-83bb-ef097168787f)
Notice the category icons missing. The System Settings menu item icon (preferences-desktop) is also missing.

See
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/merge_requests/34
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/136
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/163
https://gitlab.gnome.org/GNOME/evolution/-/issues/1698

"Regressions"
https://gitlab.gnome.org/GNOME/evolution/-/issues/1848
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/171
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/166
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/216
https://gitlab.gnome.org/GNOME/adwaita-icon-theme/-/issues/238

I can understand the GNOME developers wanting to strip down the theme and not include application specific icons, but this does seem to be contrary to the FDO Icon Theme Specification. At the very least I would argue that category icons belong in the icon set. But this seems to be yet another instance of the GNOME devs doing whatever they please, the rest of the ecosystem/specs be damned.

https://specifications.freedesktop.org/icon-naming-spec/icon-naming-spec-latest.html

There is a MR in gnome-themes-extra to ship the legacy icons, but it's been ignored for a year.
https://gitlab.gnome.org/GNOME/gnome-themes-extra/-/merge_requests/9

Not sure where we can go from here, since Adwaita no longer appears to be an FDO conformant icon theme, and I doubt any level of push-back is going to get GNOME to change course.

Does Cinnamon/Mint have a preferred/recommended alternate icon theme? Or is this one of those "up to the distro" packaging decisions?

**Comment by rcalixte** [comment]
I wasn't able to replicate this on my end so I wonder if it is something unique to EndeavourOS. Mint by default includes Adwaita and other icons sets like Breeze and Yaru. None have these issues.


> Does Cinnamon/Mint have a preferred/recommended alternate icon theme? Or is this one of those "up to the distro" packaging decisions?

I don't know if it's preferred or recommended but Mint also packages the icon themes here:
* [mint-x-icons](https://github.com/linuxmint/mint-x-icons)
* [mint-y-icons](https://github.com/linuxmint/mint-y-icons)
* [mint-l-icons](https://github.com/linuxmint/mint-l-icons)


Can you try installing the Mint icon themes from the AUR? If not, can you try to verify that it isn't something affecting just your machine or distro?

**Comment by SantiBurgos1089** [comment]
> I wasn't able to replicate this on my end so I wonder if it is something unique to EndeavourOS. Mint by default includes Adwaita and other icons sets like Breeze and Yaru. None have these issues.

Actually this is an "issue" with Arch and derivatives, they always include an upstream version of several packages, which changes what usually either Ubuntu/Linux Mint and Debian/LMDE offers, also, no other icons are offered out of the box, which might limit the icons shown at first; this also goes with themes, Adwaita as GTK and Cinnamon as Cinnamon vanilla theme are included, which might make the visual experience look bad (this doesn't happen with Manjaro since they offer several themes and icons and a pre-configured theme environment, the same as Mint/LMDE)


> I don't know if it's preferred or recommended but Mint also packages the icon themes here:
> 
>     * [mint-x-icons](https://github.com/linuxmint/mint-x-icons)
> 
>     * [mint-y-icons](https://github.com/linuxmint/mint-y-icons)
> 
>     * [mint-l-icons](https://github.com/linuxmint/mint-l-icons)

Which is the reason of why installing other themes (GTK, Cinnamon, icons) after/during setting up the environment, this is what I do @Initu-Castilhos @rcalixte 

> 
> Can you try installing the Mint icon themes from the AUR? If not, can you try to verify that it isn't something affecting just your machine or distro?

Not meant to sound rude or anything, but I am 100% sure that this will work, again and without the intention to offend or sound rude, upstream Adwaita theme in Arch and derivatives comes without too many icons available as default, GNOME may have their reasons and it might not be standard-compliant but it is their choice and there is not much to do

In case you are using Cinnamon, you can somehow ease the theming task by installing the [cinnamon-styles-artwork](https://aur.archlinux.org/packages/cinnamon-styles-artwork) package, this will pull all the required themes (cursor, GTK and icon themes) for a "Linux Mint-like experience" and to help into using the simplified settings

**Reply by rcalixte** [reply]
No problems at all.

> Actually this is an "issue" with Arch and derivatives

Have you tried opening an issue with the maintainers of EndeavourOS (or Arch) to configure different defaults for Cinnamon? I'm looking for something actionable to address this situation. I don't know what the Mint team can do about this.

> In case you are using Cinnamon, you can somehow ease the theming task by installing the [cinnamon-styles-artwork](https://aur.archlinux.org/packages/cinnamon-styles-artwork) package, this will pull all the required themes (cursor, GTK and icon themes) for a "Linux Mint-like experience" and to help into using the simplified settings

The immediate answer seems to be to switch away from using the Adwaita icon set. @Initu-Castilhos Can you try  one of the recommendations here and let us know how it works for you?

**Reply by Initu-Castilhos** [reply]
My idea is to add these missing icons to the hicolor folder and be part of cinnamon-core or create a Cinnamon icon from D.E itself.  such as mate, plasma (breeze) and Humanity (Ubuntu).

**Reply by rcalixte** [reply]
> create a Cinnamon icon from D.E itself. such as mate, plasma (breeze) and Humanity (Ubuntu).

This is what the mint-{x,y,l}-icons packages are. Did you try the suggestion from @SantiBurgos1089 to install the [cinnamon-styles-artwork](https://aur.archlinux.org/packages/cinnamon-styles-artwork) package and selecting a new theme/icon set?

> be part of cinnamon-core

Not speaking for the team but these are separate packages just like the themes and unfortunately, I don't know that that will change.

**Comment by SantiBurgos1089** [comment]
> Have you tried opening an issue with the maintainers of EndeavourOS (or Arch) to configure different defaults for Cinnamon? I'm looking for something actionable to address this situation. I don't know what the Mint team can do about this.

Regarding EndeavourOS, they provide a particular artwork, you can choose to install it or skip it on the installation process; if you don't install it, then you end up with the "upstream Arch theming" (more on that on the following paragraph)

Regarding Arch, by checking [this issue](https://github.com/linuxmint/cinnamon/issues/3575#issuecomment-374887122), coming from the package Cinnamon on upstream repositories, it seems that Clem doesn't favor the idea of shipping something particular as default (besides of what they may provide, if possible), unless every distro ships something suitable to their branding like Mint, LMDE, Manjaro and Ubuntu Cinnamon; which are the examples that comes to my head as of writing this, in those cases they provide something as artwork but doesn't necessarily means that is "the official artwork of Cinnamon creators/developers"

> The immediate answer seems to be to switch away from using the Adwaita icon set. @Initu-Castilhos Can you try one of the recommendations here and let us know how it works for you?

Upstream Arch sets the theme as Adwaita (in GTK) and Cinnamon (in Cinnamon) due to their defaults, Arch barely provides any other setting due to their KISS philosophy and it is up to the user (or derivatives) to fill that gap, so there is not much to do except to change themes manually

You can also try [mint-artwork](https://aur.archlinux.org/packages/mint-artwork) package, though that one pulls a lot of other stuffs (backgrounds, sounds and other mint-generic binaries) besides the themes

**Reply by Initu-Castilhos** [reply]
Ok
